knotx-core/src/main/java/com/cognifide/knotx/service/ServiceConfiguration.java

```diff
@@ -84,11 +84,11 @@ public void setDomain(String domain) {
             this.domain = domain;
         }
 
-        public String getPort() {
+        public Integer getPort() {
             return port;
         }
 
-        public void setPort(String port) {
+        public void setPort(Integer port) {
```

Comment:
Do we need setters on this class? How about a constructor building an immutable object instead?